src/components/[guild]/activity/ActivityLogFiltersBar/components/FiltersInput.tsx

```diff
@@ -0,0 +1,253 @@
+import {
+  Box,
+  Flex,
+  HStack,
+  IconButton,
+  Input,
+  Stack,
+  Text,
+  useColorModeValue,
+} from "@chakra-ui/react"
+import * as combobox from "@zag-js/combobox"
+import { normalizeProps, useMachine } from "@zag-js/react"
+import { useRouter } from "next/router"
+import { CaretDown, X } from "phosphor-react"
+import { KeyboardEvent, useEffect, useRef, useState } from "react"
+import {
+  isSupportedQueryParam,
+  SupportedQueryParam,
+  useActivityLogFilters,
+} from "./ActivityLogFiltersContext"
+import Dropdown from "./Dropdown"
+import FilterTag from "./FilterTag"
+import Suggestion from "./Suggestion"
+
+const FILTERS_TO_RENDER: SupportedQueryParam[] = [
+  "action",
+  "userId",
+  "roleId",
+  "rolePlatformId",
+]
+
+const getPositionerCSSVariables = (
```

Comment:
Maybe this whole function is unnecessary and we could just use getComputedStyle?

Reply:
Yup, I tried that already, but we don't actually need the whole style object (it doesn't really work if we pass it down to the `FilterTag` component), and we only need the `transform` prop when its value is not `"none"`, so I decided to write a short function which retrieves only the necessary props from computed styles. But let me know if you can think about a cleaner solution, because I'm not 100% satisfied with this one either :D